lib/DDG/Meta/Data.pm

```diff
@@ -122,6 +125,29 @@ sub get_ia {
     return $m;
 }
 
+sub _satisfy {
+	my ($ia, $by, $lookup) = @_;
+	return $by->($ia, $lookup) if ref $by eq 'CODE';
```

Comment:
@zachthompson Sorry yeah, not sure about use case for `$by`, I was describing the case for `$lookup`.

So, for example, `attribution` is a HASH reference, so simple string comparison ***is not*** enough to perform comparisons on it, you'd need a sub.